File: core/homework/homework.py
class Homework():
    def __init__(self, model):
        self.__model = model

    @property
    def homework_id(self):
        return self.__model.pk

    @property
    def year(self):
        return self.__model.year

    @property
    def name(self):
        return self.__model.name

    @property
    def total_score(self):
        return self.__model.total_score

File: view/js/api/homework.js
function HomeworkAPI(api) {
    this.create = function (year, name, totalScore, callback) {
        var data = new FormData();
        data.append("year", year);
        data.append("name", name);
        data.append("totalScore", totalScore);

        api.request("POST", "api/homework/create", data, true, callback);
    };

    this.list = function (callback) {
        var data = {};

        api.request("GET", "api/homework/list", data, true, callback);
    };

    this.upload = function (homeworkId, file, callback) {
        var data = new FormData();
        data.append("homeworkId", homeworkId);
        data.append("file", file);

        api.request("POST", "api/homework/upload", data, true, callback);
    };

    this.download = function (homeworkId, errorCallback) {
        api.downloadFile("api/homework/download", {
            token: api.accessToken,
            homeworkId: homeworkId
        }, errorCallback)
    };

    return {
        create: this.create,
        list: this.list,
        upload: this.upload,
        download: this.download
    }
}